src/cursor/cur_join.c

```diff
@@ -308,22 +331,23 @@ __curjoin_init_bloom(WT_SESSION_IMPL *session, WT_CURSOR_JOIN *cjoin,
 			if ((ret = c->next(c)) == WT_NOTFOUND)
 				goto done;
 			WT_ERR(ret);
-		}
+		} else
+			/* There should not be any other cases */
+			WT_ASSERT(session, false);
```

Comment:
I would change this to `WT_ERR(__wt_illegal_value(session, NULL));` -- it will generate a better error message if some future change introduces a bug.